### DISCUSSION TITLE
GNN regression and DAG, how to constrain which features directly influence the target values and which influence indirectly

### DISCUSSION THREAD

**Discussion by bernddude** [pyg-team/pytorch_geometric#9415] [Q&A]
Hello, i am trying to use a GAT or GCN to perform a regression 
I have a DAG which describes the influence of the features among each other and on the regression target

I understand how i can us the edge list to describe the interaction of the features 
What i am a bit at a loss is how i express the interaction between the regression target and the features ?

for example say i have three features  A, B and C and a target Y 
my DAG says that A -> C and B ->C and C -Y and B-> Y directly 

how would i set this up using the existing pytorch_geometric API ?

thank you so much for any advise , Bernd

**Comment by rusty1s** [comment]
In this case, you would need to represent your target `Y` as a readout node inside your graph, so `A`, `B`, `C` and `Y` are all nodes in your network.
